docs/cluster-providers/generic-cluster-provider.md

```diff
@@ -0,0 +1,82 @@
+# Generic Cluster Provider
```

Comment:
consider just calling this `ClusterProvider`?

Reply:
ClusterProvider is the name of the interface that it implements. I would not want change the interface name atm, it is part of our stable SPI.